src/Streamly/FileSystem/Event.hs

```diff
@@ -0,0 +1,239 @@
+#elif defined(CABAL_OS_DARWIN)
+setMetadataChanged = undefined
+#endif
+
+setModifiedFileName :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedFileName = Event.setModifiedFileName
+#elif defined(CABAL_OS_LINUX)
+setModifiedFileName s cfg = (Event.setMovedFrom s . Event.setMovedTo s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedFileName = undefined
+#endif
+
+setModifiedDirName :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedDirName = Event.setModifiedDirName
+#elif defined(CABAL_OS_LINUX)
+setModifiedDirName s cfg = (Event.setMovedFrom s . Event.setMovedTo s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedDirName = Event.setRootChanged
+#endif
+
+setModifiedSize :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedSize = Event.setModifiedSize
+#elif defined(CABAL_OS_LINUX)
+setModifiedSize s cfg = (Event.setMetadataChanged s . Event.setModified s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedSize = undefined
+#endif
+
+setModifiedLastWrite :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedLastWrite = Event.setModifiedLastWrite
+#elif defined(CABAL_OS_LINUX)
+setModifiedLastWrite s cfg =
+    (Event.setMetadataChanged s . Event.setModified s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedLastWrite = undefined
+#endif
+
+setModifiedSecurity :: Toggle -> Config -> Config
+#if defined(CABAL_OS_WINDOWS)
+setModifiedSecurity = Event.setModifiedSecurity
+#elif defined(CABAL_OS_LINUX)
+setModifiedSecurity s cfg =
+    (Event.setMetadataChanged s . Event.setModified s) cfg
+#elif defined(CABAL_OS_DARWIN)
+setModifiedSecurity = undefined
+#endif
+
+
+defaultConfig :: Config
+defaultConfig = Event.defaultConfig
+
+setAllEvents :: Toggle -> Config -> Config
+#if defined(CABAL_OS_DARWIN)
+setAllEvents = defaultConfig
+#else
+setAllEvents = Event.setAllEvents
+#endif
+
+getAbsPath :: Event -> String
+#if defined(CABAL_OS_DARWIN)
+getAbsPath = utf8ToString Event.getAbsPath
+#else
+getAbsPath ev = getRootPath ev </> getRelativePath ev
+#endif
+
+getRelativePath :: Event -> String
+#if defined(CABAL_OS_WINDOWS)
+getRelativePath = Event.getRelPath
+#elif defined(CABAL_OS_DARWIN)
+getRelativePath = getAbsPath
+#elif defined(CABAL_OS_LINUX)
+getRelativePath = utf8ToString . Event.getRelPath
+#endif
+
+getRootPath :: Event -> String
+#if defined(CABAL_OS_WINDOWS)
+getRootPath = Event.getRoot
+#elif defined(CABAL_OS_DARWIN)
+getRootPath = getAbsPath
+#elif defined(CABAL_OS_LINUX)
+getRootPath = utf8ToString . Event.getRoot
+#endif
+
+isCreated :: Event -> Bool
+isCreated =  Event.isCreated
+
+isDeleted :: Event -> Bool
+isDeleted = Event.isDeleted
+
+isMoved :: Event -> Bool
+#if defined(CABAL_OS_DARWIN)
+isMoved = Event.isMoved
+#else
+isMoved ev = Event.isMovedFrom ev || Event.isMovedTo ev
+#endif
+
+isMovedFrom :: Event -> Bool
+#if defined(CABAL_OS_DARWIN)
+isMovedFrom = Event.isMoved
+#else
+isMovedFrom = Event.isMovedFrom
+#endif
```

Comment:
Moved on darwin may not have the same semantics as movedFrom on Linux/Windows. Same holds for MovedTo.